uf3/regression/optimize.py

```diff
@@ -0,0 +1,241 @@
+    Returns:
+        columns_to_drop_2b (list): Should be passed to drop_columns argument of
+            fit_from_file
+    """
+    if original_bspline_config.leading_trim!=0:
+        raise ValueError("Currrent version is only tested for leading_trim=0")
+
+    if original_bspline_config.trailing_trim!=3:
+        raise ValueError("Currrent version is only tested for trailing_trim=3")
+
+    column_names = original_bspline_config.get_column_names()
+    interaction_partitions_num = original_bspline_config.get_interaction_partitions()[0]
+    interaction_partitions_posn = original_bspline_config.get_interaction_partitions()[1]
+
+    columns_to_drop_2b = [] 
+
+    for interaction in original_bspline_config.interactions_map[2]:
+        if not modify_2b_cutoff in original_bspline_config.knots_map[interaction]:
+            raise ValueError("Provided modify_2b_cutoff is not a knot in the %s interaction"%(str(interaction)))
+
+        num_columns_to_drop_2b = round((original_bspline_config.knots_map[interaction][-4]-modify_2b_cutoff)/knot_spacing_2b)
+
+        start_ind_2b = 1+interaction_partitions_posn[interaction]
+        end_ind_2b = start_ind_2b+interaction_partitions_num[interaction]
+        columns_to_drop_2b.extend(column_names[end_ind_2b-num_columns_to_drop_2b-3:end_ind_2b-3])
+    return columns_to_drop_2b
+
+def get_columns_to_drop_3b(original_bspline_config,
+                           modify_3b_cutoff: float,
+                           knot_spacing_3b: float):
+    """
+    Function for getting appropriate 3-body feature columns to drop for fitting
+    to lower cutoffs
+
+    Args:
+        original_bspline_config: bspline.BSplineBasis used to create the HDF5 feature file.
+            This file was produced with larger cutoff.
+        modify_3b_cutoff (float): Intended 3-body cutoff. If the 3-body interaction 
+            is A-B-C, then modify_3b_cutoff is the maximum distance between A-B (or A-C)
+            i.e the first or second term in the 3-body rmax of bspline_config.
+        knot_spacing_3b (float): knot_spacing_3b used to create the HDF5 file
+    Returns:
+        columns_to_drop_3b (list): Should be passed to drop_columns argument of
+            fit_from_file
+    """
+    if original_bspline_config.leading_trim!=0:
+        raise ValueError("Currrent version is only tested for leading_trim=0")
+
+    if original_bspline_config.trailing_trim!=3:
+        raise ValueError("Currrent version is only tested for trailing_trim=3")
+    column_names = original_bspline_config.get_column_names()
+    interaction_partitions_num = original_bspline_config.get_interaction_partitions()[0]
+    interaction_partitions_posn = original_bspline_config.get_interaction_partitions()[1]
+
+    columns_to_drop_3b = []
+    for interaction in original_bspline_config.interactions_map[3]:
+        if not modify_3b_cutoff in original_bspline_config.knots_map[interaction][0]:
```

Comment:
This is an unconventional use of the `not` operator. typically the syntax is `not in`